mysql/mysqldb.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/python
+import pymysql
+import time
+import config
+
+HOST = config.mysql['host']
+PORT = config.mysql['port']
+USER = config.mysql['user']
+PASSWORD = config.mysql['password']
+DB = config.mysql['db']
+
+def clear_all(conn):
+    with conn.cursor() as cursor:
+        sql = "DROP TABLE IF EXISTS eth_status, devp2p_hello, node_id_hash, neighbors"
+        cursor.execute(sql)
+        conn.commit()
+    print("tables dropped!")
+
+def create_all(conn):
+    create_neighbors(conn)
+    create_node_id_hash(conn)
+    create_devp2p_hello(conn)
+    create_eth_status(conn)
+
+def connect(host=HOST, port=PORT, user=USER, passwd=PASSWORD, db=DB):
+    if "CHANGEME" in host:
+        print("please change CHANGEME in config.py")
+        return None
+    try:
+        connection = pymysql.connect(host=host,port=port,user=user, passwd=passwd, db=db, charset="utf8mb4")
+        try:
+            with connection.cursor() as cursor:
+                cursor.execute("SELECT VERSION()")
+                result = cursor.fetchone()
+            # Check if anything at all is returned
+            if not result:
+                print("connected to database, but something is wrong. host={}, port={}, db={}, user={}, version='unknown'".format(host, port, db, user))
+            else:
+                print("connected to database. host={}, port={}, db={}, user={}, version={}".format(host, port, db, user, result[0]))
+        finally:
+            return connection
+    except pymysql.Error as error:
+        print("could not connect to database. host={}, port={}, db={}, user={}".format(host, port, db, user))
+    return None
+
+def create_neighbors(conn):
+    with conn.cursor() as cursor:
+        sql = "CREATE TABLE IF NOT EXISTS neighbors (" \
+              "node_id VARCHAR(128) NOT NULL, " \
+              "ip VARCHAR(39) NOT NULL, " \
+              "tcp_port SMALLINT unsigned NOT NULL, " \
+              "udp_port SMALLINT unsigned NOT NULL, " \
+              "first_ts DECIMAL(12) NULL, " \
```

Comment:
These columns are probably better named `created_at` and `updated_at`, which is what I assume they are being used for. Also, these should be TIMESTAMP field

Reply:
Also - you can add `updated_at TIMESTAMP NOT NULL DEFAULT NOW() ON UPDATE NOW()` which I believe will automatically update the timestamp

Reply:
similarly `created_at TIMESTAMP NOT NULL DEFAULT NOW()`

Reply:
I did use TIMESTAMP initially (last year), but it caused some issues, mostly about the number of decimal digits being limited. I meant to store the actual `ReceivedAt` timestamps of the packets as there may be delays to updating the database.